internal/storage/image.go

```diff
@@ -298,6 +298,36 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
 	return results, nil
 }
 
+// determines if an images is intermediate, based on ID
+func IsIntermediate(imageID string, server ImageServer) (bool, error) {
```

Comment:
Can this information be cached or stored in the storage so we avoid this calculation every time?

Reply:
could putting it in the image metadata work ?
https://github.com/cri-o/cri-o/blob/526d06b3c82c033f07210f2c0a6fa78a36ca2f0e/vendor/github.com/containers/storage/images.go#L69